### PR TITLE
feat(minifier): implement folding `substring` string fns

### PR DESCRIPTION
**Pull request by camc314**

### CODE REVIEW

**Comment by camc314** [comment]
* **#6869** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6869?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camc314 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_ecmascript/src/string_substring.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+use crate::ToInt32;
+
+pub trait StringSubstring {
+    /// `String.prototype.substring ( start , end ] )`
+    /// <https://tc39.es/ecma262/#sec-string.prototype.substring>
+    fn substring(&self, start: Option<f64>, end: Option<f64>) -> String;
```
Comment (DonIsaac) [inline crates/oxc_ecmascript/src/string_substring.rs:6]: Why not take a `u32` then have callers use `to_int_32` for casting?

Comment (camc314) [reply crates/oxc_ecmascript/src/string_substring.rs:6]: i'm open to change it -  this was just keeping the pat of the others https://github.com/oxc-project/oxc/blob/bbca7436890f6ef971807b7079f1472bc3e5a8df/crates/oxc_ecmascript/src/string_index_of.rs#L11

but as i'm thinking of this more, i wonder whether we need a const evaluation crate that accpts a JS `Value` or similar. accepting valid inputs really limits the minification we can do

Comment (Boshen) [reply crates/oxc_ecmascript/src/string_substring.rs:6]: I'll think about these later.

**Pull request merged**